internal/provider/resource_rediscloud_active_active_subscription_database.go

```diff
@@ -339,81 +344,85 @@ func resourceRedisCloudActiveActiveSubscriptionDatabaseRead(ctx context.Context,
 		return diag.FromErr(err)
 	}
 
-	if err := d.Set("support_oss_cluster_api", redis.BoolValue(db.SupportOSSClusterAPI)); err != nil {
-		return diag.FromErr(err)
-	}
-
-	if err := d.Set("external_endpoint_for_oss_cluster_api",
-		d.Get("external_endpoint_for_oss_cluster_api").(bool)); err != nil {
+	if err := d.Set("data_eviction", redis.StringValue(db.DataEvictionPolicy)); err != nil {
 		return diag.FromErr(err)
 	}
 
-	if err := d.Set("global_data_persistence", d.Get("global_data_persistence")); err != nil {
+	if err := d.Set("support_oss_cluster_api", redis.BoolValue(db.SupportOSSClusterAPI)); err != nil {
 		return diag.FromErr(err)
 	}
 
-	if err := d.Set("global_alert", d.Get("global_alert")); err != nil {
+	if err := d.Set("external_endpoint_for_oss_cluster_api", redis.BoolValue(db.UseExternalEndpointForOSSClusterAPI)); err != nil {
 		return diag.FromErr(err)
 	}
 
-	if err := d.Set("global_password", d.Get("global_password")); err != nil {
+	if err := d.Set("memory_limit_in_gb", redis.Float64(*db.CrdbDatabases[0].MemoryLimitInGB)); err != nil {
 		return diag.FromErr(err)
 	}
 
-	if err := d.Set("global_source_ips", d.Get("global_source_ips")); err != nil {
+	if err := d.Set("enable_tls", redis.BoolValue(db.CrdbDatabases[0].Security.EnableTls)); err != nil {
 		return diag.FromErr(err)
 	}
 
-	var region_db_configs []map[string]interface{}
-	public_endpoint_config := make(map[string]interface{})
-	private_endpoint_config := make(map[string]interface{})
-	for _, region_db := range db.CrdbDatabases {
+	var regionDbConfigs []map[string]interface{}
+	publicEndpointConfig := make(map[string]interface{})
+	privateEndpointConfig := make(map[string]interface{})
+	for _, regionDb := range db.CrdbDatabases {
+		// Set the endpoints for the region
+		publicEndpointConfig[redis.StringValue(regionDb.Region)] = redis.StringValue(regionDb.PublicEndpoint)
+		privateEndpointConfig[redis.StringValue(regionDb.Region)] = redis.StringValue(regionDb.PrivateEndpoint)
+		// Check if the region is in the state as an override
+		x := getStateOverrideRegion(d, redis.StringValue(regionDb.Region))
```

Comment:
Can we come up with a more reasonable name on the variable? x is not that descriptive...

Reply:
Good spot, definitely, I'll update that now